src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -942,52 +951,54 @@ ep_buffer_manager_write_event (
 	// Before we pick a buffer, make sure the event is enabled.
 	ep_return_false_if_nok (ep_event_is_enabled (ep_event));
 
+	// Check to see an event thread was specified. If not, then use the current thread.
+	if (event_thread == NULL)
+		event_thread = thread;
+
+	EventPipeThread *current_thread;
+	current_thread = ep_thread_get ();
+	ep_raise_error_if_nok (current_thread != NULL);
+
+	// session_state won't be freed if write_in_progress is set.
+	EP_ASSERT (ep_thread_is_rundown_thread (current_thread) || ep_thread_get_session_write_in_progress (current_thread) == ep_session_get_index (session));
```

Comment:
Is there anything that prevents the rundown thread from setting the write_in_progress flag? Assuming there isn't I think its simpler to set the flag rather than complicate the conditionals.